packages/react-virtual/src/index.tsx

```diff
@@ -47,7 +47,8 @@ function useVirtualizerBase<
   }, [])
 
   useIsomorphicLayoutEffect(() => {
-    return instance._willUpdate()
+    instance._willUpdate()
+    instance.measureElement = instance.measureElement.bind(instance)
```

Comment:
Kinda feels like this is more core issue not react specific, hmm what about something like

As we should have elements in cache, we can observe elements when we set scroll element
 
<img width="1659" alt="Zrzut ekranu 2024-08-29 o 11 47 58" src="https://github.com/user-attachments/assets/90d33c21-5077-474a-a2eb-ed029d911182">

Reply:
Hmm, since `this.cleanup()` gets called in `_willUpdate`, `this.elementsCache` ends up being empty... It seems necessary to call `cleanup` when `this.scrollElement` changes, and it feels natural to clear the `elementsCache` within `cleanup`, but I can't figure out what the best solution is...

How about making it event-driven and only calling _measureElement after _willUpdate has been called?
I thought so, but it would be better if cleanup() also handled event listeners.

Hmm,I can’t come up with a good idea.

Reply:
True, this is bit tricky as when clearing scrollElement we can set it back, but with elementsCache we can't. Kinda feels maybe just stop observing with observer.disconnect() and observe back. 

> How about making it event-driven and only calling _measureElement after _willUpdate has been called?

When we observe back elements then _measureElement will be called from RO callback.

Reply:
Maybe for now let's not clear the cache on clear and observe it back later https://github.com/TanStack/virtual/pull/830